cookbooks/aws-parallelcluster-scheduler-plugin/resources/execute_event_handler.rb

```diff
@@ -212,4 +211,12 @@ def command_with_retries(command, retries, user, cwd, env)
 
     cmd
   end
+
+  def raise_and_write_chef_error(cmd, resource, event_log_out, event_log_err)
```

Comment:
let's be more generic: the function can be defined in the helper.rb and it accepts only the message as parameter.
Then the message is built in the caller and passed to this function.